ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go

```diff
@@ -1155,22 +1111,334 @@ func generateNetworkLatencyTestCases(name, expectedHappyResponseBody string) []n
 				}, nil)
 			},
 		},
+		{
+			name:               "failed-to-unmarshal-json",
+			expectedStatusCode: 500,
+			requestBody: map[string]interface{}{
+				"DelayMilliseconds":  delayMilliseconds,
+				"JitterMilliseconds": jitterMilliseconds,
+				"Sources":            ipSources,
+				"Unknown":            "",
+			},
+			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("failed to check existing network fault: failed to unmarshal tc command output: unexpected end of JSON input. TaskArn: taskArn"),
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(happyTaskResponse, nil)
+			},
+			setExecExpectations: func(exec *mock_execwrapper.MockExec, ctrl *gomock.Controller) {
+				ctx, cancel := context.WithTimeout(context.Background(), requestTimeoutDuration)
+				mockCMD := mock_execwrapper.NewMockCmd(ctrl)
+				exec.EXPECT().NewExecContextWithTimeout(gomock.Any(), gomock.Any()).Times(1).Return(ctx, cancel)
+				exec.EXPECT().CommandContext(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(mockCMD)
+				mockCMD.EXPECT().CombinedOutput().Times(1).Return([]byte("["), nil)
+			},
+		},
+		{
+			name:               "os/exec-times-out",
+			expectedStatusCode: 500,
+			requestBody: map[string]interface{}{
+				"DelayMilliseconds":  delayMilliseconds,
+				"JitterMilliseconds": jitterMilliseconds,
+				"Sources":            ipSources,
+				"Unknown":            "",
+			},
+			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("failed to check existing network fault: 'nsenter --net=/some/path tc -j q show dev eth0 parent 1:1' command failed with the following error: 'signal: killed'. std output: ''. TaskArn: taskArn"),
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(happyTaskResponse, nil)
+			},
+			setExecExpectations: func(exec *mock_execwrapper.MockExec, ctrl *gomock.Controller) {
+				ctx, cancel := context.WithTimeout(context.Background(), requestTimeoutDuration)
+				mockCMD := mock_execwrapper.NewMockCmd(ctrl)
+				exec.EXPECT().NewExecContextWithTimeout(gomock.Any(), gomock.Any()).Times(1).Return(ctx, cancel)
+				exec.EXPECT().CommandContext(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(mockCMD)
+				mockCMD.EXPECT().CombinedOutput().Times(1).Return([]byte{}, errors.New("signal: killed"))
+			},
+		},
+		{
+			name:                 "request timed out",
```

Comment:
Could you help me understand why this `tcCommandEmptyOutput` will lead to a request time out?

Reply:
This is because when checking for timeout, we don't look at the error returned by os/exec. Instead, we look at ctx.Err(). In this test case, ctx is initialized with `context.WithTimeout(context.Background(), -1*time.Second)`, since we used -1 as timeout, when checking its error, it will always be timeout

Reply:
Thanks.

IMO we can use `Do()` to mock the timeout case where we can sleep for some time longer than the requestTimeout.

ie  
```
exec.EXPECT().NewExecContextWithTimeout(gomock.Any(), gomock.Any()).Do(func(i, j interface{}){
...// sleep
}).Return(ctx, cancel)
```

Reply:
SGTM. Addressing this in the next PR